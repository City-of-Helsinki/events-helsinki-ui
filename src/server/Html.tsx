import React from "react";

interface Props {
  assets: {
    css: string[];
    js: string[];
  };
  content: string;
  helmet: {
    meta: {
      toComponent: () => void;
    };
    title: {
      toComponent: () => void;
    };
  };
  state: object;
}

const Html: React.FC<Props> = ({ assets, content, helmet, state }) => {
  return (
    <html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1, shrink-to-fit=no"
        />
        <meta name="theme-color" content="#000000" />
        <link rel="manifest" href="/manifest.json" />
        <link rel="shortcut icon" href="/favicon.ico" />
        {helmet.meta.toComponent()}
        {helmet.title.toComponent()}
        {assets.css &&
          assets.css.map((c: string, idx: number) => (
            <link key={idx} href={c} rel="stylesheet" />
          ))}
      </head>

      <body>
        <noscript>You need to enable JavaScript to run this app.</noscript>
        <div id="root" dangerouslySetInnerHTML={{ __html: content }} />
        <script
          dangerouslySetInnerHTML={{
            __html: `window.__APOLLO_STATE__=${JSON.stringify(state).replace(
              /</g,
              "\\u003c"
            )};`
          }}
        />
        {assets.js &&
          assets.js.map((j: string, idx: number) => (
            <script key={idx} src={j} />
          ))}
      </body>
    </html>
  );
};

export default Html;
