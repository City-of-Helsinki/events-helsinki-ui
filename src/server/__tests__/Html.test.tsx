import * as React from 'react';
import renderer from 'react-test-renderer';

import Html from '../Html';

test('Html matches snapshot', () => {
  const assets = {
    css: ['test1.css', 'test2.css'],
    js: ['test1.js', 'test2.js'],
  };
  const content = '<p>Test content</p>';
  const canonicalUrl = 'http://localhost:3000';
  // Helmet.renderStatic() is not available on client side so use mock data to test
  const helmet = {
    link: {
      toComponent: () => `<link rel="canonical" href="${canonicalUrl}" />`,
    },
    meta: {
      toComponent: () =>
        '<meta data-react-helmet="true" name="description" content="testing react helmet">',
    },
    title: {
      toComponent: () => '<title>Test title</title>',
    },
  };

  const component = renderer.create(
    <Html
      assets={assets}
      content={content}
      helmet={helmet}
      state={{}}
      canonicalUrl={canonicalUrl}
      initialI18nStore={{
        en: {},
        fi: {},
        sv: {},
      }}
      initialLanguage="fi"
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
