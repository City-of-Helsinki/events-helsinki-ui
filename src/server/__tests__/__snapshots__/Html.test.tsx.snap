// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Html matches snapshot 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charSet="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
      name="viewport"
    />
    <meta
      content="#000000"
      name="theme-color"
    />
    <link
      href="/manifest.json"
      rel="manifest"
    />
    <link
      href="/favicon.ico"
      rel="shortcut icon"
    />
    &lt;meta data-react-helmet="true" name="description" content="testing react helmet"&gt;
    &lt;title&gt;Test title&lt;/title&gt;
    <link
      href="test1.css"
      rel="stylesheet"
    />
    <link
      href="test2.css"
      rel="stylesheet"
    />
  </head>
  <body>
    <noscript>
      You need to enable JavaScript to run this app.
    </noscript>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Test content</p>",
        }
      }
      id="root"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window.__APOLLO_STATE__={};",
        }
      }
    />
    <script
      src="test1.js"
    />
    <script
      src="test2.js"
    />
  </body>
</html>
`;
